main.py

```diff
@@ -0,0 +1,262 @@
+        st.experimental_rerun()
+
+
+def welcome_page():
+    header()
+    logged_in_as()
+
+    st.session_state.current_page = "welcome"
+
+    create_game_button = st.button("Create Game")
+    st.write("\n")
+    entered_game_code = str.upper(st.text_input("Enter 6-digit game code"))
+    join_game_button = st.button("Join Game")
+
+    if create_game_button:
+        new_game_code = wcs.generate_game_code()
+        st.session_state.game_code = new_game_code
+        table_columns = ["GAME_ID", "GAME_CODE", "GAME_TIME_START", "IS_ACTIVE"]
+        table_values = [str(uuid.uuid4()), st.session_state.game_code, dt.datetime.now(), True]
+        snow.push("GAMES", table_columns, table_values)
+
+        st.session_state.current_page = "lobby"
+        st.experimental_rerun()
+
+    if join_game_button or entered_game_code:
+        query = f"""
+        select * from {DB_NAME}.{SCHEMA_NAME}.GAMES
+        where IS_ACTIVE = TRUE
+        """
+
+        df_games = snow.pull(query)
+        is_active_game = entered_game_code in df_games.GAME_CODE.tolist()
+
+        if is_active_game:
+            st.session_state.current_page = "lobby"
+            st.session_state.game_code = entered_game_code
+            st.experimental_rerun()
+
+        else:
+            st.error("Not an active game code.")
+
+    back_widget(to="login")
+
+
+def join_lobby(df_lobby):
+    game_code, player_name = st.session_state.game_code, st.session_state.player_name
+
+    if [game_code, player_name] in df_lobby[["GAME_CODE", "PLAYER_NAME"]].values.tolist():
+        pass
+
+    else:
+        colnames = ["game_code", "player_name", "joined_at"]
+        values = [game_code, player_name, dt.datetime.now()]
+        snow.push("lobby_info", colnames, values)
+
+    st.session_state.in_lobby = True
+
+
+def lobby_page():
+    header()
+
+    query = f"""
+    select * from {DB_NAME}.{SCHEMA_NAME}.lobby_info
+    where game_code = '{st.session_state.game_code}'
+    order by joined_at
+    """
+    df_lobby = snow.pull(query)
+
+    n_players = df_lobby.shape[0]
+
+    c1, c2 = st.columns(2)
+    with c1:
+        st.write(f"### code: {st.session_state.game_code}")
+    with c2:
+        st.write(f"### {n_players} / 10")
+
+    join_lobby(df_lobby)
+    
+    for name in df_lobby.PLAYER_NAME:
+        # st.write(f"{name} \u2713")
+        st.write(name)
+
+    ## start game widget
+    start_game_button = st.button("start game")
+
+    if start_game_button:
+        if n_players < 3:
+            st.error("Not enough players")
+            
+            ### HACK
+            if st.session_state.player_name == "dev":
+                st.session_state.current_page = "game start"
+                st.experimental_rerun()
+            ###
+        
+        else:
+            st.session_state.current_page = "game start"
+            st.experimental_rerun()
+
+
+    back_widget()
+
+
+def game_start_page():
+    header()
+
+    back_widget(to="lobby")
```

Comment:
NOTE: this is where game instructions should be included

Also, it should like to a page (function) called play_round_page() that implements the functionality of app.py